sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1093,6 +1126,16 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
 
         .. versionadded:: 0.23
 
+    interaction_cst : list of lists or sets of int, default=None
```

Comment:
Could we just say "list of iterable" or "list of array-like" or something like that? And same in the error messages.

Seems a bit unnecessary to special-case for sets?

Reply:
`iterable of iterables of int, default=None`